edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/DeviceScopeIdentitiesCache.cs

```diff
@@ -155,7 +162,7 @@ async Task RefreshCache()
                 {
                     Events.StartingRefreshCycle();
                     var currentCacheIds = new List<string>();
-                    IServiceIdentitiesIterator iterator = this.serviceProxy.GetServiceIdentitiesIterator();
+                    IServiceIdentitiesIterator iterator = this.serviceProxy.GetNestedServiceIdentitiesIterator();
                     while (iterator.HasNext)
                     {
                         IEnumerable<ServiceIdentity> batch = await iterator.GetNext();
```

Comment:
>IEnumerable<ServiceIdentity> batch = await iterator.GetNext(); [](start = 24, length = 62)

From David: maybe instead of blocking on individual network calls, we collate a collection of tasks, and try to resolve them all in parallel.